### ISSUE TITLE
`comma-spacing` and `space-before-function-paren` are conflict in type

### ISSUE THREAD

**Issue by Austaras** [typescript-eslint/typescript-eslint#707]
Don't know what's the root cause of this problem, I guess it's the parser

**What code were you trying to parse?**
```ts
function x<R extends Record<string, (this: any) => any>>(fns: R) {
    return fns
}
```
when eslint config includes
```json
"comma-spacing": "warn",
"space-before-function-paren": ["warn", "never"],
```

**What did you expect to happen?**
code shall pass lint

**What actually happened?**
code violates rule `comma-spacing` when
```ts
function x<R extends Record<string,(this: any) => any>>(fns: R) {
    return fns
}
```
and violates `space-before-function-paren` when
```ts
function x<R extends Record<string, (this: any) => any>>(fns: R) {
    return fns
}
```

**Versions**

| package                     | version |
| --------------------------- | ------- |
| `@typescript-eslint/parser` | `1.12.0` |
| `TypeScript`                | `3.5.3` |
| `ESLint`                    | `6.0.2` |

**Comment by bradzacher** [comment]
Same issue as in #364, except it's a different rule.

**Comment by Austaras** [comment]
well, thanks for the explanation, hop it will get fixed soon

**Comment by bradzacher** [comment]
`comma-spacing` has not been fixed yet

**Comment by Austaras** [comment]
IMO `comma-spacing` doesn't need a fix since generic parameters are supposed to have a space after comma(like `prettier` does) so as you said this rule "just work"

**Comment by bradzacher** [comment]
sounds good!
